src/main/java/com/google/sps/servlets/TasksServlet.java

```diff
@@ -61,13 +62,73 @@ public TasksServlet(
     this.tasksClientFactory = tasksClientFactory;
   }
 
-  private List<Task> getTasks(TasksClient tasksClient) throws IOException {
+  /**
+   * Returns Tasks from the user's Tasks account
+   *
+   * @param request Http request from client. Should contain idToken and accessToken
+   * @param response 403 if user is not authenticated, list of Tasks otherwise
+   * @param googleCredential a valid google credential object (already verified)
+   * @throws IOException if an issue arises while processing the request
+   */
+  @Override
+  public void doGet(
+      HttpServletRequest request, HttpServletResponse response, Credential googleCredential)
+      throws IOException {
+    assert googleCredential != null
+        : "Null credentials (i.e. unauthenticated requests) should already be handled";
+
+    // Get tasks from Google Tasks
+    TasksClient tasksClient = tasksClientFactory.getTasksClient(googleCredential);
     List<TaskList> taskLists = tasksClient.listTaskLists();
-    List<Task> tasks = new ArrayList<>();
-    for (TaskList taskList : taskLists) {
-      tasks.addAll(tasksClient.listTasks(taskList));
+    List<Task> tasks = getTasks(tasksClient);
+
+    // Initialize Tasks Response
+    List<String> taskListTitles = getTaskListTitles(taskLists);
+    int tasksToComplete = getTasksToComplete(tasks);
+    int tasksDueToday = getTasksDueToday(tasks);
+    int tasksCompletedToday = getTasksCompletedToday(tasks);
+    int tasksOverdue = getTasksOverdue(tasks);
+    TasksResponse tasksResponse =
+        new TasksResponse(
+            taskListTitles, tasksToComplete, tasksDueToday, tasksCompletedToday, tasksOverdue);
+
+    // Convert tasks to JSON and print to response
+    JsonUtility.sendJson(response, tasksResponse);
+  }
+
+  /**
+   * Add a new task object to a tasklist
+   *
+   * @param request HTTP request from client. Must contain a taskListId. Body must contain a task
+   *     entity (https://developers.google.com/tasks/v1/reference/tasks)
+   * @param response Http response to be sent to client
+   * @param googleCredential valid, verified google credential object
+   * @throws IOException if an issue occurs with the tasksService
+   */
+  @Override
+  public void doPost(
+      HttpServletRequest request, HttpServletResponse response, Credential googleCredential)
+      throws IOException {
+    assert googleCredential != null
+        : "Null credentials (i.e. unauthenticated requests) should already be handled";
+    TasksClient tasksClient = tasksClientFactory.getTasksClient(googleCredential);
+
+    String taskListId = request.getParameter("taskListId");
+
+    if (taskListId == null || taskListId.equals("")) {
+      response.sendError(400, "taskListId must be present in request");
+      return;
+    }
+
+    Gson gson = new Gson();
+    Task taskToPost = gson.fromJson(request.getReader(), Task.class);
```

Comment:
What happens if the body is empty? Or if the task can't be parsed?

(Maybe you don't need any special handling, but at least add a test to show what the behavior is.)

Reply:
I added some handling for this. Thanks for catching this!